### PR TITLE
Update to NuGet version 4.3.0-preview1-2500

### PR DESCRIPTION
**Pull request by nkolev92**
Update to NuGet version 4.3.0-preview1-2479

### CODE REVIEW

**Comment by rohit21agrawal** [comment]
@nguerrera @dsplaisted any idea why the tests are failing with this error? 

```
error MSB4018: The "ResolvePackageDependencies" task failed unexpectedly. [D:\j\workspace\release_windo---82602456\bin\Release\Tests\CleanTargetRemovesAll\AppsWithFrameworkReferences\StopwatchLib\StopwatchLib.csproj] [D:\j\workspace\release_windo---82602456\build\build.proj]
```

**Comment by nguerrera** [comment]
I'm pulling this down now to take a look at why things are failing.

**Comment by nguerrera** [comment]
The assembly version has changed from 4.3.0.0 to 4.3.0.2479 and yet again, we can't load a different nuget dll than what's in the msbuild directory of CLI: https://github.com/dotnet/cli/issues/4214

**Comment by nguerrera** [comment]
I tried changing things to replace CLI nuget bits with the ones we're using in .dotnet, but there are breaking changes here and CLI's dotnet restore fails if I do that. 

Can you update CLI first, then SDK?

**Comment by emgarten** [comment]
> The assembly version has changed from 4.3.0.0 to 4.3.0.2479 and yet again

@nkolev92 we were including the 4th build number in the strong name before?

**Comment by nguerrera** [comment]
> we were including the 4th build number in the strong name before?

Looks like the opposite. Earlier versions did not have 4th number in strong name, this update does.

However, even if you revert the strong name version change,  we'll likely run up against breaking changes when we load the version sitting in the CLI's msbuild folder.

File: build/DependencyVersions.props
Comment (nguerrera) [inline build/DependencyVersions.props]: We can remove all of the SideBySide goop now that these versions are aligned. :)

**Comment by nguerrera** [comment]
Still, with the build number in strong name, it means we're going to hit this on every update even if there are no breaking changes!

**Review by emgarten — APPROVED** [review]

**Comment by rohit21agrawal** [comment]
@nguerrera  i have updated sdk with a build of nuget that doesn't contain build numbers in strong name, and yet the tests are failing with similar errors as before.

**Comment by livarcocc** [comment]
@rohit21agrawal I believe you need to update the CLI first, right?

**Comment by rohit21agrawal** [comment]
@livarcocc not sure, i can try that. so far we have always updated the sdk first, but i'll give it a shot.

**Comment by rohit21agrawal** [comment]
have sent out a PR at https://github.com/dotnet/cli/pull/6287

**Comment by rohit21agrawal** [comment]
@livarcocc @nguerrera can we get a build of sdk with the tests disabled? we will try inserting the sdk into cli to see if it unblocks it.

**Comment by emgarten** [comment]
🔔  how can we get this moving along? We need to start regularly integrating NuGet into CLI/SDK. Currently it looks like this is blocked by having merged some assemblies on the NuGet side, the issues look due to the way the SDK tests using older bits where the assemblies are not yet merged.

**Comment by dsplaisted** [comment]
@emgarten Let's try it like this:

- Rebase on top of the latest changes in the repo
- Push these changes to a separate branch of the SDK repo
- We'll kick off a VSO build to create a package with the updated SDK
- The build of the SDK can then be inserted into the CLI

Since this skips the SDK tests, I'd also like to see them run locally before we insert the updated SDK into the CLI.

**Comment by zhili1208** [comment]
rebased and updated the nuget version to 4.3.0-preview1-2500

**Comment by zhili1208** [comment]
@dsplaisted I rebased and pushed changes to nkolev92/sdk repo, Can you kick off a vso build from that repo? or I need to create an repo under dotnet/sdk?

**Comment by zhili1208** [comment]
@livarcocc @dsplaisted ping...

**Comment by dsplaisted** [comment]
To do a VSO build the changes need to be in the dotnet/sdk repo.  I've pushed them to the [update-nuget](https://github.com/dotnet/sdk/tree/update-nuget) branch and kicked off a build.

**Comment by dsplaisted** [comment]
@zhili1208 @nkolev92 The version number of the package from the VSO build is Microsoft.NET.Sdk.2.0.0-preview2-20170425-7.  You can use this to insert into the CLI.

**Comment by zhili1208** [comment]
@livarcocc I don't have permission to update the cli PR. Could you give me the permission?

**Comment by dsplaisted** [comment]
@zhili1208 It looks like you need to join the Microsoft OSS org: https://opensource.microsoft.com/link

**Comment by zhili1208** [comment]
@dsplaisted I just joined the Microsoft OSS, but still don't have write permission to cli repo

**Comment by livarcocc** [comment]
@dotnet-bot Test this please

**Comment by zhili1208** [comment]
NuGet  4.3.0-preview1-2500 merged to VS branch lab/vsuwpt

**Comment by zhili1208** [comment]
@livarcocc when this PR can be merged to SDK repo?

**Comment by nguerrera** [comment]
If it's merged to CLI, then next step is to update DotnetCLIVersion.txt, that should unblock the tests.

**Comment by nkolev92** [comment]
@livarcocc @nguerrera 
I have updated the PR, but the tests are still failing.
Can you please take a look?

**Comment by nguerrera** [comment]
I'll take a look shortly.

**Comment by nguerrera** [comment]
I think I've fixed this. Replacing with #1169

**Pull request closed without merging**
